cmd/nerdctl/create_linux_test.go

```diff
@@ -33,3 +35,68 @@ func TestCreate(t *testing.T) {
 	base.Cmd("start", tID).AssertOK()
 	base.Cmd("logs", tID).AssertOutContains("foo")
 }
+
+func TestCreateWithMACAddress(t *testing.T) {
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+	networkBridge := "testNetworkBridge" + tID
+	networkMACvlan := "testNetworkMACvlan" + tID
+	networkIPvlan := "testNetworkIPvlan" + tID
+	base.Cmd("network", "create", networkBridge, "--driver", "bridge").AssertOK()
+	base.Cmd("network", "create", networkMACvlan, "--driver", "macvlan").AssertOK()
+	base.Cmd("network", "create", networkIPvlan, "--driver", "ipvlan").AssertOK()
+	t.Cleanup(func() {
+		base.Cmd("network", "rm", networkBridge).AssertOK()
+		base.Cmd("network", "rm", networkMACvlan).AssertOK()
+		base.Cmd("network", "rm", networkIPvlan).AssertOK()
+	})
+	tests := []struct {
+		Network string
+		Err     string
+	}{
+		{"host", "conflicting options"},
+		{"none", "can't open '/sys/class/net/eth0/address'"},
+		{"container:whatever" + tID, "conflicting options"},
+		{"bridge", ""},
+		{networkBridge, ""},
+		{networkMACvlan, ""},
+		{networkIPvlan, "not support"},
+	}
+	for i, test := range tests {
+		containerName := fmt.Sprintf("%s_%d", tID, i)
+		macAddress, err := nettestutil.GenerateMACAddress()
+		if err != nil {
+			t.Errorf("failed to generate MAC address: %s", err)
+		}
+		cmd := base.Cmd("create", "--network", test.Network, "--mac-address", macAddress, "--name", containerName, testutil.CommonImage, "cat", "/sys/class/net/eth0/address")
+		if test.Err == "" || test.Network == "none" {
```

Comment:
check `none` in else ?

Reply:
SGTM, I'll combine it with wantErr just follow what @Zheaoli advised